pkg/controller/status/conditions.go

```diff
@@ -0,0 +1,119 @@
+package status
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// OperatorDisabled defines the condition type when the operator's primary function has been disabled
+	OperatorDisabled configv1.ClusterStatusConditionType = "Disabled"
+	// InsightsUploadDegraded defines the condition type (when set to True) when an archive can't be successfully uploaded
+	InsightsUploadDegraded configv1.ClusterStatusConditionType = "UploadDegraded"
+	// InsightsDownloadDegraded defines the condition type (when set to True) when the Insights report can't be successfully downloaded
+	InsightsDownloadDegraded configv1.ClusterStatusConditionType = "InsightsDownloadDegraded"
+)
+
+type conditionsMap map[configv1.ClusterStatusConditionType]configv1.ClusterOperatorStatusCondition
+
+type conditions struct {
+	entryMap conditionsMap
+}
+
+func newConditions(cos *configv1.ClusterOperatorStatus, time metav1.Time) *conditions {
+	entries := map[configv1.ClusterStatusConditionType]configv1.ClusterOperatorStatusCondition{
+		configv1.OperatorAvailable: {
+			Type:               configv1.OperatorAvailable,
+			Status:             configv1.ConditionUnknown,
+			LastTransitionTime: time,
+			Reason:             "",
+		},
+		configv1.OperatorProgressing: {
+			Type:               configv1.OperatorProgressing,
+			Status:             configv1.ConditionUnknown,
+			LastTransitionTime: time,
+			Reason:             "",
+		},
+		configv1.OperatorDegraded: {
+			Type:               configv1.OperatorDegraded,
+			Status:             configv1.ConditionUnknown,
+			LastTransitionTime: time,
+			Reason:             "",
+		},
+		configv1.OperatorUpgradeable: {
+			Type:               configv1.OperatorUpgradeable,
```

Comment:
I think we didn't have this condition before and I think it's never set so we can probably remove it.